src/StmtToViz.cpp

```diff
@@ -0,0 +1,2900 @@
+        IRVisitor::visit(op);
+        std::vector<const IRNode *> args;
+        for (const auto &arg : op->vectors) {
+            args.push_back(arg.get());
+        }
+        set_compute_costs(op, 0, args);
+        set_data_costs(op, op->type.bits() * op->type.lanes(), args);
+    }
+
+    void visit(const VectorReduce *op) override {
+        IRVisitor::visit(op);
+        const int factor = op->value.type().lanes() / op->type.lanes();
+        const int op_count = op->type.lanes() * (factor - 1);
+        set_compute_costs(op, op_count, {op->value.get()});
+        set_data_costs(op, 0, {op->value.get()});
+    }
+
+    void visit(const LetStmt *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, 0, {op->value.get(), op->body.get()}, {op->value.get()});
+        set_data_costs(op, 0, {op->value.get(), op->body.get()}, {op->value.get()});
+    }
+
+    void visit(const AssertStmt *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, 1, {op->condition.get()});
+        set_data_costs(op, 0, {op->condition.get()});
+    }
+
+    void visit(const ProducerConsumer *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, 0, {op->body.get()}, {});
+        set_data_costs(op, 0, {op->body.get()}, {});
+    }
+
+    void visit(const For *op) override {
+        // The cost of a loop-node essentially depends on its iteration
+        // count. The cost model currently ignores such costs.
+        IRVisitor::visit(op);
+        set_compute_costs(op, 0, {op->min.get(), op->extent.get(), op->body.get()}, {op->min.get(), op->extent.get()});
+        set_data_costs(op, 0, {op->min.get(), op->extent.get(), op->body.get()}, {op->min.get(), op->extent.get()});
+    }
+
+    void visit(const Acquire *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, 1, {op->semaphore.get(), op->count.get(), op->body.get()}, {op->semaphore.get(), op->count.get()});
+        set_data_costs(op, 0, {op->semaphore.get(), op->count.get(), op->body.get()}, {op->semaphore.get(), op->count.get()});
+    }
+
+    void visit(const Store *op) override {
+        IRVisitor::visit(op);
+        set_compute_costs(op, 0, {op->predicate.get(), op->value.get(), op->index.get()});
+        set_data_costs(op, op->value.type().bits() * op->value.type().lanes(), {op->predicate.get(), op->value.get(), op->index.get()});
+    }
+
+    void visit(const Provide *op) override {
+        IRVisitor::visit(op);
+        std::vector<const IRNode *> args;
+        for (const auto &arg : op->values) {
+            args.push_back(arg.get());
+        }
+        for (const auto &arg : op->args) {
+            args.push_back(arg.get());
+        }
+        args.push_back(op->predicate.get());
+        set_compute_costs(op, 0, args, {});
+        set_data_costs(op, 0, args, {});
+    }
+
+    void visit(const Allocate *op) override {
+        // We do not model allocation/de-allocation costs
```

Comment:
Do we think we will need to at some point? Does this require a TODO or a tracking issue?